### DISCUSSION TITLE
How to model a feed of child features

### DISCUSSION THREAD

**Discussion by ahbou** [pointfreeco/swift-composable-architecture#2136] [Q&A]
I have a feed of child features mixed together let's call it `FeedSection`.
It's like a facebook newsfeed where each item could be a news article, a video or an ad.

Let's say the parent is called Home and it holds an `IdentifiedArrayOf<FeedSection.State>` 

I took inspiration from the `Destination` Reducer so my `FeedSection` looks like this:

```
public struct FeedSection: Reducer {
    public enum State: Equatable, Identifiable {
        public var id: UUID {
            switch self {
            case let .news(state):
                return state.id
            case let .ad(state):
                return state.id
            case let .video(state):
                return state.id
            }
        }

        case news(News.State)
        case ad(Ad.State)
        case video(Carousel.State)
    }

    public enum Action: Equatable {
        case news(News.Action)
        case ad(Ad.Action)
        case video(Video.Action)
    }

    public var body: some ReducerOf<Self> {
        EmptyReducer()
            .ifCaseLet(/State.news, action: /Action.base) {
                News()
            }
            .ifCaseLet(/State.ad, action: /Action.brands) {
                Ad()
            }
            .ifCaseLet(/State.video, action: /Action.carousel) {
                Video()
            }
    }
}

public struct FeedSectionView: View {
    let store: StoreOf<FeedSection>

    init(store: StoreOf<FeedSection>) {
        self.store = store
    }

    public var body: some View {
        SwitchStore(self.store) {
            CaseLet(state: /FeedSection.State.news, action: FeedSection.Action.news) { store in
                NewsView(store: store)
            }
            CaseLet(state: /FeedSection.State.ad, action: FeedSection.Action.ad) { store in
               AdView(store: store)
            }
            CaseLet(state: /FeedSection.State.video, action: FeedSection.Action.video) { store in
               VideoView(store: store)
            }
        }
    }
}

```

This works and Home renders all the sections but I get runtime warning about missing child.

Is this the right approach to model a feed of child features?

**Comment by stephencelis** [comment]
@ahbou I'm not sure we have enough info to troubleshoot. Can you share the exact runtime warning you're getting and what causes it? Or can you provide enough code for us to run and repro?

**Reply by ahbou** [reply]
I'm setting up a sample project. But today I realized that `SwitchStore` can only have up to 9 cases.
The example I gave above is simplified but in reality we have a dozen of feed sections.
Is there a workout or are we modelling the feed wrong?

**Reply by ahbou** [reply]
I ended up fixing it by using 2 `SwitchStore` and a Default with an empty view on the one that has the least elements.
But it still makes me wonder if the right way to model a feed using TCA is a state enum?

**Reply by mbrandonw** [reply]
Hi @ahbou, without commenting on whether or not it's the best way to handle "feed"-like state, I can suggest a slight improvement. If you are targeting the newest version of the library, 0.53, then there is a simpler way to handle the `switch`. The details are spelled out in the [release notes](https://github.com/pointfreeco/swift-composable-architecture/releases/tag/0.53.0). This allows you to switch over any number of enum cases.

**Reply by ahbou** [reply]
Thanks @mbrandonw! I'm using the `prerelease` branch but the updates to `SwitchStore` are there as well.
I got rid of my workaround 👍 
Please do comment if you think there's a better way to model a feed state :)
